src/qt/bantablemodel.cpp

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) 2011-2015 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "bantablemodel.h"
+
+#include "clientmodel.h"
+#include "guiconstants.h"
+#include "guiutil.h"
+
+#include "net.h"
+#include "sync.h"
+#include "utiltime.h"
+
+#include <QDebug>
+#include <QList>
+#include <QTimer>
+
+#include <boost/date_time/posix_time/posix_time.hpp>
+#include <boost/date_time/c_local_time_adjustor.hpp>
+
+// private implementation
+class BanTablePriv
+{
+public:
+    /** Local cache of peer information */
+    QList<CCombinedBan> cachedBanlist;
+    /** Column to sort nodes by */
+    int sortColumn;
+    /** Order (ascending or descending) to sort nodes by */
+    Qt::SortOrder sortOrder;
+
+    /** Pull a full list of banned nodes from CNode into our cache */
+    void refreshBanlist()
+    {
+        std::map<CSubNet, int64_t> banMap;
```

Comment:
Why would it make it more complicated? Worse comes to worse, you comment it, which is not really needed anyways.